### ISSUE TITLE
WinDbg begginers question: Unable to acquire .NET debugger interface. Error: 8027025b

### ISSUE THREAD

**Issue by arisjr** [rodneyviana/netext#13]
I am getting the following error on WinDbg Preview 1,0,2007.0.6001:

**!windex -tree
Unable to acquire .NET debugger interface. Error: 8027025b
c0000005 Exception in netext.windex debugger extension.
      PC: 5a07691c  VA: 00000000  R/W: 0  Parameter: 00000000**

.chain
Extension DLL search Path:
 C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Common Files\Apple\Apple Application Support;C:\Program Files (x86)\Common Files\Apple\Mobile Device Support;C:\Program Files\Common Files\Apple\Apple Application Support;C:\Program Files\Common Files\Apple\Mobile Device Support;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\SafeNet\Authentication\SAC\x64;C:\Program Files\SafeNet\Authentication\SAC\x32;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Users\aristeu.agaj\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\
Extension DLL chain:
    netext: image 2.1.57.5000, API 1.0.0, built Mon May 18 13:26:11 2020
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\netext.dll]
    mscorwks: image 2.0.50727.9151, built Sat Mar 21 02:10:51 2020
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\mscorwks.dll]
    ELFBinComposition: image 10.0.20153.1000, API 0.0.0, 
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\winext\ELFBinComposition.dll]
    dbghelp: image 10.0.20153.1000, API 10.0.6, 
        [path: C:\Users\Desktop\memdump\WinDbg\x86\dbghelp.dll]
    ext: image 10.0.20153.1000, API 1.0.0, 
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\winext\ext.dll]
    exts: image 10.0.20153.1000, API 1.0.0, 
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\WINXP\exts.dll]
    uext: image 10.0.20153.1000, API 1.0.0, 
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\winext\uext.dll]
    ntsdexts: image 10.0.20153.1000, API 1.0.0, 
        [path: C:\Users\aristeu.agaj\Desktop\memdump\WinDbg\x86\WINXP\ntsdexts.dll]

**Comment by rodneyviana** [comment]
In the debugger, what is the output of:
```
.cordll -u -ve -l
```

**Comment by rodneyviana** [comment]
Idle for a long time

**Comment by rodneyviana** [comment]
@acherkashin 

Please help me understand the problem better:

Are you able to get heap information from sos (like '!dumpheap -stat')?
What's the result of NetExt's '!wver' ?
What do you get from '.cordll -ve -u -l' ?

**Comment by acherkashin** [comment]
> Are you able to get heap information from sos (like '!dumpheap -stat')?

yes

> What's the result of NetExt's '!wver' ?

0:000> !wve
No export wve found

> What do you get from '.cordll -ve -u -l' ?

0:000> .cordll -ve -u -l
CLR DLL status: No load attempts

**Comment by rodneyviana** [comment]
You used "!wve" instead of "!wver"
The response to '.cordll -ve -u -l' suggests that there is no .NET runtime available in the image. So, if you are debugging .NET core, most times the process where the code run in not w3wp.exe but an external process.
If you have problem with '!wver', check 'lmv m core*' and 'lm v m clr'

**Comment by TreaHorton** [comment]
Hi Rodney, I am seeing the same behavior and another wrinkle:

- If I try with a trace that does not have CLR DLL loaded, then I get the same error as mentioned above which makes sense:

  0:000> .cordll -u -ve -l
  CLRDLL: No CLR image loaded (i.e. mscorwks.dll)
  CLR DLL status: No load attempts
  
  0:000> !wver
  CLRDLL: No CLR image loaded (i.e. mscorwks.dll)
  Unable to acquire .NET debugger interface
  Try running .cordll -l. Error: 8027025b

- You're saying that is expected correct?

- But for some traces where I do have CLR DLL loaded, I am getting an exception that crashes windbgx:

0:000> .cordll -u -ve -l
CLRDLL: C:\Windows\Microsoft.NET\Framework64\v4.0.30319\mscordacwks.dll:4.8.9032.00 f:8
doesn't match desired version 4.8.4430.00 f:8
WARNING: Unable to register for CLR module notifications
CLRDLL: Loaded DLL d:\symbols\mscordacwks_AMD64_AMD64_4.8.4430.00.dll\61270E85ac1000\mscordacwks_AMD64_AMD64_4.8.4430.00.dll
Automatically loaded SOS Extension
CLR DLL status: Loaded DLL d:\symbols\mscordacwks_AMD64_AMD64_4.8.4430.00.dll\61270E85ac1000\mscordacwks_AMD64_AMD64_4.8.4430.00.dll

0:000> !wver
Runtime(s) Found: 1
0: Filename: mscordacwks_Amd64_Amd64_4.8.4430.00.dll 
+--> .NET 4.8.4430.0

NetExt (this extension) Version: 2.1.65.5000

0:000> !windex
Starting indexing at 10:20:45 AM
c00000fd Exception in netext.windex debugger extension.
      PC: 00007ffa`d932f485  VA: 000000a9`28585fc8  R/W: 1  Parameter: 00000000`00000000

**Comment by TreaHorton** [comment]
I suspect this might be expected as well if there's no managed heap in the trace etc. ?

**Comment by acherkashin** [comment]
@rodneyviana Sorry for bothering you, but could you guide me what should I do to get it working?

**Comment by rodneyviana** [comment]
If .cordll is not load the DAC DLL it means the heap is in a unstable state or not initialized.
Are you sure sos core is working? What do you get when you run sos' eeheap?
```
!eeheap
```

**Comment by rodneyviana** [comment]
> I suspect this might be expected as well if there's no managed heap in the trace etc. ?
> 
> (I mean, not the crash, but an exception if there's nothing to load)

You are correct. It seems to be the case. If sos doesn’t work, NetExt will not work either as they use the same DAC API

**Comment by acherkashin** [comment]
I get the following results when run `!eeheap`

![image](https://user-images.githubusercontent.com/9947582/170733813-277ba449-b721-4f19-a8b6-f26f8ff1a97f.png)

Regular commands work fine, e.g. `!dumpheap -stat`, `!do 00007ff976e22660`, but commands specific to the extension do not work, e.g. `!wdo 00007ff976e22660`

**Comment by rodneyviana** [comment]
> I get the following results when run `!eeheap`
> 
> ![image](https://user-images.githubusercontent.com/9947582/170733813-277ba449-b721-4f19-a8b6-f26f8ff1a97f.png)
> 
> Regular commands work fine, e.g. `!dumpheap -stat`, `!do 00007ff976e22660`, but commands specific to the extension do not work, e.g. `!wdo 00007ff976e22660`

I would need to analyze the dump to see what’s happening.

**Comment by acherkashin** [comment]
I guess the problem is not in the dump, I have dump that I could analyze previously and currently it doesn't work for some reason.

**Comment by rodneyviana** [comment]
stale discussion
